src/preferences.tsx

```diff
@@ -21,55 +21,55 @@ export interface IPreferences {
   urlBasename: string;
 }
 
-const { condaStoreConfig = {} } =
-  typeof window !== "undefined" && (window as any);
+let condaStoreConfig: any = {};
+if (typeof window !== "undefined" && "condaStoreConfig" in window) {
+  condaStoreConfig = window.condaStoreConfig;
+}
 
 export const prefDefault: Readonly<IPreferences> = {
   apiUrl:
-    process.env.REACT_APP_API_URL ??
     condaStoreConfig.REACT_APP_API_URL ??
+    process.env.REACT_APP_API_URL ??
     "http://localhost:8080/conda-store/",
 
   authMethod:
-    (process.env.REACT_APP_AUTH_METHOD as IPreferences["authMethod"]) ??
     (condaStoreConfig.REACT_APP_AUTH_METHOD as IPreferences["authMethod"]) ??
+    (process.env.REACT_APP_AUTH_METHOD as IPreferences["authMethod"]) ??
     "cookie",
 
   authToken:
-    process.env.REACT_APP_AUTH_TOKEN ??
     condaStoreConfig.REACT_APP_AUTH_TOKEN ??
+    process.env.REACT_APP_AUTH_TOKEN ??
     "",
 
   loginUrl:
-    process.env.REACT_APP_LOGIN_PAGE_URL ??
     condaStoreConfig.REACT_APP_LOGIN_PAGE_URL ??
+    process.env.REACT_APP_LOGIN_PAGE_URL ??
     "http://localhost:8080/conda-store/login?next=",
 
   styleType:
-    process.env.REACT_APP_STYLE_TYPE ??
     condaStoreConfig.REACT_APP_STYLE_TYPE ??
+    process.env.REACT_APP_STYLE_TYPE ??
     "green-accent",
 
-  showAuthButton: process.env.REACT_APP_SHOW_AUTH_BUTTON
-    ? JSON.parse(process.env.REACT_APP_SHOW_AUTH_BUTTON)
-    : condaStoreConfig !== undefined &&
-      condaStoreConfig.REACT_APP_SHOW_AUTH_BUTTON !== undefined
-    ? JSON.parse(condaStoreConfig.REACT_APP_SHOW_AUTH_BUTTON)
-    : true,
+  showAuthButton:
+    (condaStoreConfig.REACT_APP_SHOW_AUTH_BUTTON ??
+      process.env.REACT_APP_SHOW_AUTH_BUTTON ??
+      "true") === "true",
```

Comment:
I'd really prefer one and only one way of configuring this. Allowing multiple ways of setting a boolean is how we get the [Norway problem](https://www.bram.us/2022/01/11/yaml-the-norway-problem/) with yaml, for example, and it does basically nothing to improve readability IMO.

Since we're on the topic, why are we doing string comparisons here anyway? I assume because `process.env` does no parsing of env variables, and treats them all as strings (because that's what they actually are)?

If you want to make it case insensitive that sounds useful, though.

Reply:
- Yes, all process.env values are strings
- re: Norway problem. Hehe, that's funny, hadn't heard of that one until today. The essence of that problem as I understand it, though, is allowing unquoted strings, not necessarily about having different ways to represent the same type. 
- But I agree: case-insensitive match on "true" should be good enough